-1

**Failed Tests**:   Build 
**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ded78f/40349/summary.html
**COMMIT**: cb20f3cfacdcab3180e4b40c80c6acc0d4e6cb99
**CMSSW**: CMSSW_14_1_X_2024-07-11-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/45433/40349/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre><artificial>:(.text+0x2c62): undefined reference to `char const* edm::typelookup::className<DTKeyedConfigContainerRcd>()'
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02845/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: <artificial>:(.text+0x2c6a): undefined reference to `std::type_info const& edm::typelookup::classTypeInfo<DTKeyedConfigContainerRcd>()'
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02845/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: tmp/el8_amd64_gcc12/src/CondCore/BaseKeyedPlugins/src/CondCoreBaseKeyedPlugins/cclLCdgm.ltrans0.ltrans.o: in function `ProductResolverWrapper<DTKeyedConfigListRcd, cond::persistency::KeyList, cond::DefaultInitializer<cond::persistency::KeyList> >::recordKey() const [clone .localalias]':
<artificial>:(.text+0x2c82): undefined reference to `char const* edm::typelookup::className<DTKeyedConfigListRcd>()'
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02845/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/../lib/gcc/x86_64-redhat-linux-gnu/12.3.1/../../../../x86_64-redhat-linux-gnu/bin/ld.bfd: <artificial>:(.text+0x2c8a): undefined reference to `std::type_info const& edm::typelookup::classTypeInfo<DTKeyedConfigListRcd>()'
collect2: error: ld returned 1 exit status
gmake: *** [tmp/el8_amd64_gcc12/src/CondCore/BaseKeyedPlugins/src/CondCoreBaseKeyedPlugins/libCondCoreBaseKeyedPlugins.so] Error 1
Leaving library rule at CondCore/BaseKeyedPlugins
>> Leaving Package CondCore/BaseKeyedPlugins
>> Package CondCore/BaseKeyedPlugins built
>> Entering Package CondCore/BTauPlugins
</pre>

--- a/CondCore/BaseKeyedPlugins/BuildFile.xml
+++ b/CondCore/BaseKeyedPlugins/BuildFile.xml
@@ -1,4 +1,7 @@
+<use name="CondCore/CondDB"/>
 <use name="CondCore/ESSources"/>
 <use name="CondFormats/Calibration"/>
+<use name="CondFormats/Common"/>
+<use name="CondFormats/DataRecord"/>
 <use name="CondFormats/DTObjects"/>
 <flags EDM_PLUGIN="1"/>

